File: frontend/lib/models/customer.dart
class Customer {
  final String id;
  final String? boutiqueId;
  final String fullName;
  final String? email;
  final String? phone;
  final String? address;
  final String? city;
  final String? governorate;
  final String? postalCode;
  final String? country;
  final String? notes;
  final int totalOrders;
  final double totalSpent;
  final String? lastOrderDate;
  final String? createdAt;
  final String? updatedAt;

  Customer({
    required this.id,
    this.boutiqueId,
    required this.fullName,
    this.email,
    this.phone,
    this.address,
    this.city,
    this.governorate,
    this.postalCode,
    this.country,
    this.notes,
    this.totalOrders = 0,
    this.totalSpent = 0.0,
    this.lastOrderDate,
    this.createdAt,
    this.updatedAt,
  });

  factory Customer.fromJson(Map<String, dynamic> json) => Customer(
    id: json['id'].toString(),
    boutiqueId: json['boutiqueId']?.toString(),
    fullName: json['fullName'] ?? '',
    email: json['email'],
    phone: json['phone'],
    address: json['address'],
    city: json['city'],
    governorate: json['governorate'],
    postalCode: json['postalCode'],
    country: json['country'],
    notes: json['notes'],
    totalOrders: json['totalOrders'] ?? 0,
    totalSpent: (json['totalSpent'] ?? 0).toDouble(),
    lastOrderDate: json['lastOrderDate'],
    createdAt: json['createdAt'],
    updatedAt: json['updatedAt'],
  );
}

File: backend/src/main/java/io/makewebsite/entity/Customer.java
package io.makewebsite.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "customers", indexes = {
    @Index(name = "idx_customers_boutique_email", columnList = "boutique_id, email"),
    @Index(name = "idx_customers_boutique", columnList = "boutique_id")
})
public class Customer {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(updatable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "boutique_id", nullable = false)
    private Boutique boutique;

    @NotNull
    @Size(max = 100)
    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Size(max = 150)
    private String email;

    @Size(max = 20)
    private String phone;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Size(max = 100)
    private String city;

    @Size(max = 100)
    private String governorate;

    @Size(max = 20)
    @Column(name = "postal_code")
    private String postalCode;

    @Size(max = 100)
    private String country;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Builder.Default
    @Column(name = "total_orders")
    private int totalOrders = 0;

    @Builder.Default
    @Column(name = "total_spent", precision = 12, scale = 2)
    private BigDecimal totalSpent = BigDecimal.ZERO;

    @Column(name = "last_order_date")
    private LocalDateTime lastOrderDate;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
